tests/modules/s3_datasets/test_dataset_table.py

```diff
@@ -242,3 +256,152 @@ def test_delete_table(client, table, dataset_fixture, db, group):
         tableUri=table_to_delete.tableUri,
     )
```

Comment:
nit: it would be nice to either add another test for `delete_table_with_filters` and assert that the filters are deleted in `lf_client.delete_table_data_filter`

Reply:
added another test for this case